phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -673,16 +668,21 @@ private boolean extractTablesAndIndexes(PhoenixConnection conn) {
         return indexInfos;
     }
 
+    @VisibleForTesting
+    public String getViewSql(String tableName, String schemaName) {
+       return "SELECT DISTINCT COLUMN_FAMILY, TENANT_ID FROM "
+                + "SYSTEM.CHILD_LINK "
+                + "WHERE TABLE_NAME = \'" + tableName + "\' "
+                + (!StringUtil.EMPTY_STRING.equals(schemaName) ? "AND TABLE_SCHEM = \'"
```

Comment:
That seems like a very long way to check emptiness of a string. Perhaps add an isEmpty() fn in StringUtil class, if schemaName.isEmpty() is not sufficient for whatever reason.

Reply:
Also, would be much simpler to write as schemaName.isEmpty() ? "" : "AND TABLE_SCHEM = \'" + schemaName + "\' " .. instead of negating it first.

Reply:
Please add spaces before AND's

Reply:
to avoid any NPE due to schemaName being null, it's better to check .equals on a constant. It reads longer but it is not that long :)

Reply:
There are spaces after the quote, but before AND makes it more readable. ll change it.